### PR TITLE
 PRT-167-Data-reliability-for-rest-tendermint-grpc-optional-params

### PR DESCRIPTION
**Pull request by candostyavuz**
The requested block has been added to the `newChainMessage`, replacing the prior usage of `spectypes.NOT_APPLICABLE` for both REST and gRPC interfaces.

### CODE REVIEW

File: protocol/chainlib/chainproxy/rpcInterfaceMessages/restMessage_test.go
Comment (ranlavanet) [inline protocol/chainlib/chainproxy/rpcInterfaceMessages/restMessage_test.go:17]: you dont need to do this check you can do 
require.Nil(t, params)

Comment (ranlavanet) [reply protocol/chainlib/chainproxy/rpcInterfaceMessages/restMessage_test.go:17]: no need for print as well the require is enough.

File: protocol/chainlib/common_test.go
Comment (ranlavanet) [inline protocol/chainlib/common_test.go:120]: lets not leave commented code in PR

File: protocol/chainlib/grpc.go
Comment (ranlavanet) [inline protocol/chainlib/grpc.go:82]: we don't have URI in grpc, no need to overwrite block parsing

File: protocol/chainlib/rest.go
Comment (ranlavanet) [inline protocol/chainlib/rest.go:77]: Here as well no need to overwrite block parsing. this is only for tendermint. URI calls

File: protocol/chainlib/rest_test.go
Comment (ranlavanet) [inline protocol/chainlib/rest_test.go:116]: if Spec path is not used in this test you don't need to specify it, it will just be empty by default.

File: protocol/parser/parser.go
Comment (ranlavanet) [inline protocol/parser/parser.go:310]: this is not the right way to do it, if grpc has objects unmarshalled not as strings which is why I assume you need it changed you can have a method to check the type and convert only the necessary and known types
```
convertInterfaceToString(input interface{}) (string, error) {
switch inp := input .(type) {
	case string:
		return info
	case int:
	       return strconv.formatInt(info,10)
	default:
	       return utils.LavaFormatError("Failed to convert to string, unknown type", info: input , inp  etc..)
}
```

Comment (ranlavanet) [reply protocol/parser/parser.go:310]: that way we know which types we support and which we do not

File: protocol/rpcprovider/rewardserver/reward_server.go
Comment (ranlavanet) [inline protocol/rpcprovider/rewardserver/reward_server.go:76]: this should be a function, that gets these as inputs and returns them unified that way no mistakes can happen in the order.

File: protocol/rpcprovider/rewardserver/reward_server.go
Comment (ranlavanet) [inline protocol/rpcprovider/rewardserver/reward_server.go:107]: as well here

**Review by ranlavanet — CHANGES_REQUESTED** [review]

File: protocol/chainlib/chainproxy/rpcInterfaceMessages/restMessage_test.go
Comment (candostyavuz) [reply protocol/chainlib/chainproxy/rpcInterfaceMessages/restMessage_test.go]: fixed in : fd83ad126719d12e3ec38756fd82b04c61f4af1d

File: protocol/chainlib/common_test.go
Comment (candostyavuz) [reply protocol/chainlib/common_test.go]: fixed: fd83ad126719d12e3ec38756fd82b04c61f4af1d

File: protocol/chainlib/grpc.go
Comment (candostyavuz) [reply protocol/chainlib/grpc.go]: fixed: 524cb1833ba85f59885c6f6e829b7f9d6d08e451

File: protocol/chainlib/rest.go
Comment (candostyavuz) [reply protocol/chainlib/rest.go]: fixed: 524cb1833ba85f59885c6f6e829b7f9d6d08e451

File: protocol/parser/parser.go
Comment (candostyavuz) [reply protocol/parser/parser.go]: Yes the reason was grpc being unmarshalled not as strings. Fixed here: c261e1ecef87053813dee7dc1761a4ff6fe5b86c

File: protocol/rpcprovider/rewardserver/reward_server.go
Comment (candostyavuz) [reply protocol/rpcprovider/rewardserver/reward_server.go]: fixed: 57dff5307fc3945fee74060280ef65ceaa19e079

File: protocol/rpcprovider/rewardserver/reward_server.go
Comment (candostyavuz) [reply protocol/rpcprovider/rewardserver/reward_server.go]: fixed: https://github.com/lavanet/lava/commit/57dff5307fc3945fee74060280ef65ceaa19e079

File: protocol/parser/parser.go
Comment (ranlavanet) [inline protocol/parser/parser.go:433]: there is a better way to convert float64 to string use strconv.FormatFloat() instead.

**Review by ranlavanet — CHANGES_REQUESTED** [review]

File: protocol/chainlib/rest_test.go
Comment (candostyavuz) [reply protocol/chainlib/rest_test.go:116]: It is actually used in the unit tests because `ParseMsg()` assigns `SpechPath` to `msg` parameter, so it needs to be specified here as well.

File: protocol/parser/parser.go
Comment (candostyavuz) [reply protocol/parser/parser.go]: Sure: 41f80bad125e2f5f36fc678183fe4c806426151e

File: protocol/chainlib/chainproxy/rpcInterfaceMessages/restMessage_test.go
Comment (ranlavanet) [inline protocol/chainlib/chainproxy/rpcInterfaceMessages/restMessage_test.go:12]: eth has no rest messages so it might be confusing ethereum uses only jsonrpc

**Review by ranlavanet — CHANGES_REQUESTED** [review]

File: protocol/chainlib/chainproxy/rpcInterfaceMessages/restMessage_test.go
Comment (candostyavuz) [reply protocol/chainlib/chainproxy/rpcInterfaceMessages/restMessage_test.go]: Fixed: ae1eabb9451c17943a95cdbe083634adc9dfb073

**Review by ranlavanet — APPROVED** [review]

**Comment by ranlavanet** [comment]
waiting for @omerlavanet approval as well before merge

**Pull request merged**
